code/__HELPERS/text.dm

```diff
@@ -53,6 +53,36 @@
 /proc/sanitize(t,list/repl_chars = null)
 	return html_encode(sanitize_simple(t,repl_chars))
 
+//Used for preprocessing entered text
+/proc/sanitize_rp(var/input, var/max_length = MAX_MESSAGE_LEN, var/encode = 1, var/trim = 1, var/extra = 1)
+	if(!input)
+		return
+
+	if(max_length)
+		input = copytext(input,1,max_length)
```

Comment:
I think if we need to trim we should be be loud rather than silently dropping parts of a long emote. Additionally, we're calling copytext here with MAX_MESSAGE_LEN, and then the caller does it **again** to what we return, which seems fairly redundant?